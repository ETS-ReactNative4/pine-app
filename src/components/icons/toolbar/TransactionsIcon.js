import React, { Component } from 'react';
import { StyleSheet, View, Image } from 'react-native';
import PropTypes from 'prop-types';

const IMAGE_DEFAULT = require('../../../images/icons/toolbar/Transactions.png');
const IMAGE_WHITE = require('../../../images/icons/toolbar/TransactionsWhite.png');

const styles = StyleSheet.create({
  icon: {
    width: 50,
    height: 50
  }
});

export default class TransactionsIcon extends Component {
  render() {
    const image = this.props.white ? IMAGE_WHITE : IMAGE_DEFAULT;

    return (
      <View style={this.props.style}>
        <Image source={image} style={styles.icon} />
      </View>
    );
  }
}

TransactionsIcon.propTypes = {
  style: PropTypes.any,
  white: PropTypes.bool
};
