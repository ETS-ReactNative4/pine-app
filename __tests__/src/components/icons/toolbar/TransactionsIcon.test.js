import React from 'react';
import renderer from 'react-test-renderer';
import TransactionsIcon from '../../../../../src/components/icons/toolbar/TransactionsIcon';

describe('TransactionsIcon', () => {
  it('renders default icon correctly', () => {
    const tree = renderer.create(
      <TransactionsIcon />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders white icon correctly', () => {
    const tree = renderer.create(
      <TransactionsIcon white={true} />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
