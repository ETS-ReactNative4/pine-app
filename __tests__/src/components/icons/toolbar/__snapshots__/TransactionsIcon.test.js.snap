// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TransactionsIcon renders default icon correctly 1`] = `
<View>
  <Image
    source={
      Object {
        "testUri": "../../../src/images/icons/toolbar/Transactions.png",
      }
    }
    style={
      Object {
        "height": 50,
        "width": 50,
      }
    }
  />
</View>
`;

exports[`TransactionsIcon renders white icon correctly 1`] = `
<View>
  <Image
    source={
      Object {
        "testUri": "../../../src/images/icons/toolbar/TransactionsWhite.png",
      }
    }
    style={
      Object {
        "height": 50,
        "width": 50,
      }
    }
  />
</View>
`;
